src/main/java/net/sf/jabref/gui/maintable/MainTableSelectionListener.java

```diff
@@ -105,25 +105,16 @@ public void listChanged(ListEvent<BibEntry> e) {
         if (!enabled) {
             return;
         }
+
         EventList<BibEntry> selected = e.getSourceList();
-        BibEntry newSelected = null;
-        while (e.next()) {
-            if (e.getType() == ListEvent.INSERT) {
```

Comment:
`ListEvent.INSERT` means that the entry was inserted into the selection (thus selected by the user), and there lies the problem.
When multiple entries were selected and you select an single entry inside that selection there are only `ListEvent.DELETE`-events, thus JabRef doesn't recognize the change.